xarray/tests/test_merge.py

```diff
@@ -272,7 +282,7 @@ def test_merge_no_conflicts_multi_var(self):
 
     def test_merge_no_conflicts_preserve_attrs(self):
         data = xr.Dataset({"x": ([], 0, {"foo": "bar"})})
-        actual = xr.merge([data, data])
+        actual = xr.merge([data, data], combine_attrs="no_conflicts")
```

Comment:
does this make sense? I think the `no_conflicts` refers to `compat` and not `combine_attrs`, but otherwise this will keep failing.

#4749 suggested to change the default merge strategy to `drop_conflicts`, so we could revert all these tiny test changes after that (but maybe we need to deprecate the current default first?)

Reply:
is the old default "override"?

Reply:
yes. This is done by `compat`, which delegates to `fillna`, which in turn calls `apply_ufunc`. This raises the bigger question whether `compat` should affect `attrs`, or whether it would be better to completely separate `compat` from `combine_attrs` / `keep_attrs`.

Reply:
actually, the result of `compat` seems to always be to either raise for conflicting attributes (`"identical"`) or to keep the first object's `attrs` (`"override"`). Completely separating `compat` won't work entirely because `compat="identical"` will still raise. Not sure what to do about that – should we deprecate it?